Osiris/Hacks/Misc.cpp

```diff
@@ -506,3 +506,64 @@ void Misc::killMessage(GameEvent& event) noexcept
     cmd += "\"";
     interfaces.engine->clientCmdUnrestricted(cmd.c_str());
 }
+
+void Misc::drawBombDamage() noexcept
+{
+    const auto localPlayer = interfaces.entityList->getEntity(interfaces.engine->getLocalPlayer());
+
+    //No Alive return since it is useful if you want to call it out to a mate that he will die
+    if (!localPlayer || !config.misc.bombDamage)
+        return;
+
+    for (int i = interfaces.engine->getMaxClients(); i <= interfaces.entityList->getHighestEntityIndex(); i++)
+    {
+        auto entity = interfaces.entityList->getEntity(i);
+        if (!entity || entity->isDormant() || entity->getClientClass()->classId != ClassId::PlantedC4 || !entity->c4Ticking())
+            continue;
+
+        auto vecBombDistance = entity->origin() - localPlayer->origin();
+
+        const auto a = 450.7f;
+        const auto b = 75.68f;
+        const auto c = 789.2f;
+        const auto d = (vecBombDistance.length() - b) / c;
+        auto flDamage = a * exp(-d * d);
+
+        const auto ArmorValue = localPlayer->armor();
+        const auto flArmorRatio = 0.5f;
+        const auto flArmorBonus = 0.5f;
+        if (ArmorValue > 0) {
+            auto flNew = flDamage * flArmorRatio;
+            auto flArmor = (flDamage - flNew) * flArmorBonus;
+
+            if (flArmor > static_cast<float>(ArmorValue)) {
+                flArmor = static_cast<float>(ArmorValue)* (1.f / flArmorBonus);
+                flNew = flDamage - flArmor;
+            }
+
+            flDamage = flNew;
+        }
+        const auto bombDamage = max((int)ceilf(flDamage), 0);
```

Comment:
inconsistance casting. C-style casting right here but in other places static_cast is used.
wouldn't flooring an int be faster than flooring a float?
^ convert to int then floor

Reply:
Casting it to an int before flooring it would result in a wrong result since it will round down.

Reply:
i failed to consider the ceiling operation was dependant on the fact it was a float.
it should still be changed to a <static_cast> instead of a C style cast, better consistancy and is checked by the compiler.